### PR TITLE
Fix: string params starting with '0'

### PR DESCRIPTION
**Pull request by prenx4x**
*Issue #, if available:*
[200](https://github.com/awslabs/serverless-application-model/issues/200)

*Why is this change necessary?*
Templates with string params leading with '0' are converted to integers. This patch fixes this issues.

*How does it address the issue?*
Check output template and put quotes around such params

*What side effects does this change have?*

*Did you change a dependency in `requirements/base.txt`?*
    *If so, did you run `make update-reproducible-reqs`*

*Checklist:*

- [ ] Write Design Document ([Do I need to write a design document?](https://github.com/awslabs/aws-sam-cli/blob/develop/DEVELOPMENT_GUIDE.rst#design-document))
- [ ] Write unit tests
- [ ] Write/update functional tests
- [x] Write/update integration tests
- [x] `make pr` passes
- [ ] Write documentation

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: tests/regression/deploy/test_deploy_regression.py [hunk 1]
```diff
@@ -67,6 +67,7 @@ def test_deploy_with_all_args(self, template_file):
             "aws_stack_name": aws_stack_name,
             "sam_stack_name": sam_stack_name,
             "capabilities": "CAPABILITY_IAM",
+            "s3_bucket": self.s3_bucket.name,
```
Comment (jfuss) [inline tests/regression/deploy/test_deploy_regression.py:70]: Curious, why did you add these? Trying to understand why this change required these to be added to the tests.

File: tests/regression/package/test_package_regression.py [hunk 2]
```diff
@@ -26,9 +26,7 @@ def tearDown(self):
             "aws-appsync-graphqlschema.yaml",
             "aws-appsync-resolver.yaml",
             "aws-appsync-functionconfiguration.yaml",
-            "aws-apigateway-restapi.yaml",
```
Comment (jfuss) [inline tests/regression/package/test_package_regression.py:29]: Can you explain why these were removed?

It's ok to remove them but I want to ensure removing them does not loose coverage in testing unexpectedly.

File: tests/integration/testdata/package/aws-serverlessrepo-application.yaml [hunk 3]
```diff
@@ -12,5 +12,5 @@ Metadata:
     ReadmeUrl: ./README.md
     Labels: ['tests']
     HomePageUrl: https://github.com/user1/my-app-project
-    SemanticVersion: 0.0.1
+    SemanticVersion: 1.0.1
```
Comment (jfuss) [inline tests/integration/testdata/package/aws-serverlessrepo-application.yaml:15]: Why shouldn't we leave this as `0.0.1`?

File: tests/regression/deploy/test_deploy_regression.py (hunk 1 above)
Comment (prenx4x) [reply tests/regression/deploy/test_deploy_regression.py:70]: So when I was working on this, we had the resolve-s3 feature merged, which needed either s3_bucket or -resolve-s3 parameter for deploy. But since last week, I pushed another PR which removed this check in deploy as it was not needed in deploy (only needed in package), which got merged yesterday. Thus I can now remove this as we don't need it.

File: tests/regression/package/test_package_regression.py (hunk 2 above)
Comment (prenx4x) [reply tests/regression/package/test_package_regression.py:29]: For this test, the hash generated inside SAM has a leading zero which is causing it to fail. I could maybe modify the value of the property a bit so we have a different hash (hopefully without leading zeros) and see if that works.

File: tests/integration/testdata/package/aws-serverlessrepo-application.yaml (hunk 3 above)
Comment (prenx4x) [reply tests/integration/testdata/package/aws-serverlessrepo-application.yaml:15]: this is failing because our output is '0.0.1' and awscli's output is 0.0.1 which is failing. Thus I changed the leading 0 to 1 so we don't wrap it in quotes.

File: tests/regression/package/test_package_regression.py (hunk 2 above)
Comment (prenx4x) [reply tests/regression/package/test_package_regression.py]: Tried this and modified the SemanticVersion property. Now there is no value generated with leading zero. As far as I can tell, this changes does not affect the crux of the tests.

File: tests/integration/testdata/package/samconfig.toml [hunk 4]
```diff
@@ -0,0 +1,10 @@
+version = 0.1
+[default]
+[default.deploy]
+[default.deploy.parameters]
+stack_name = "test-deploy-guided-set-parameter-0-aws-serverless-function-yaml-0-0-0"
```
Comment (jfuss) [inline tests/integration/testdata/package/samconfig.toml:5]: I don't think we should be committing this. The current tests should not require this. Committing this, runs the risk of other tests using these defaults and failing. 

For example, there can only be one S3 Bucket with a given name (names are global for S3). Hardcoding a bucket (event in a config file like this), will cause seemingly random failures for our package tests if someone other than you tries to run the tests.

Comment (prenx4x) [reply tests/integration/testdata/package/samconfig.toml]: thats right. this file shouldn't be here.

**Review by jfuss — APPROVED** [review]
